### PR TITLE
fix(gpu): fix memory error in cg classical PBS

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_cg_classic.cuh [hunk 1]
```diff
@@ -67,12 +67,12 @@ __global__ void device_programmable_bootstrap_cg(
...
-  Torus *accumulator =
-      (Torus *)accumulator_fft +
-      (ptrdiff_t)(sizeof(double2) * polynomial_size / 2 / sizeof(Torus));
+  Torus *accumulator = (Torus *)selected_memory;
   Torus *accumulator_rotated =
-      (Torus *)accumulator + (ptrdiff_t)polynomial_size;
+      (Torus *)accumulator + (ptrdiff_t)(polynomial_size);
+  double2 *accumulator_fft =
+      (double2 *)(accumulator_rotated) +
+      (ptrdiff_t)(polynomial_size * sizeof(Torus) / sizeof(double2));
```
Comment (IceTDrinker) [inline backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_cg_classic.cuh:75]: maybe a naive question but does that work alignment wise ?

Comment (IceTDrinker) [reply backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_cg_classic.cuh:75]: I would have thought double2 (128 bits, but could be aligned on 64 bits) requiring potentially more alignment than Torus (32 or 64 bits)

Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_cg_classic.cuh:75]: Yes, we padded the accumulator to be sure.

Comment (IceTDrinker) [reply backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_cg_classic.cuh:75]: ok !

**Pull request merged**